onnx/defs/reduction/utils.cc

```diff
@@ -103,16 +100,14 @@ to `False` instead of `True`.)DOC";
         return;
       }
 
-      int64_t keep_dims = 1, noop_with_empty_axes = 0;
-      auto attr_proto = ctx.getAttribute("keepdims");
-      if (attr_proto) {
-        keep_dims = attr_proto->i();
-      }
-      auto noop_attr_proto = ctx.getAttribute("noop_with_empty_axes");
-      if (noop_attr_proto) {
-        noop_with_empty_axes = noop_attr_proto->i();
-      }
+      const auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+      const int64_t input_ndim = input_shape.dim_size();
+      auto output_shape = ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
```

Comment:
The above statement has the side-effect of initializing the output-shape to be an empty shape. That could be a problem in cases where the return happens early because the output rank cannot be determined (such as new line number 119 below, old line number 124). In protobuf, there is a difference between the shape not being set (means unknown rank) and the shape being initialized to an empty shape (means rank 0 scalar).

Reply:
Ah ok